test-suite/src/test/scala/org/scalajs/testsuite/library/UnionTypeTest.scala

```diff
@@ -0,0 +1,153 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2015, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.library
+
+import scala.language.implicitConversions
+
+import scala.scalajs.js
+import js.|
+
+import org.scalajs.jasminetest.JasmineTest
+import org.scalajs.testsuite.Typechecking._
+
+import js.JSConverters._
+
+object UnionTypeTest extends JasmineTest {
+
+  private implicit def unionAsJSAny(x: _ | _): js.Any = x.asInstanceOf[js.Any]
```

Comment:
Ah, I see... that makes sense. But can we add "converters" for `String`, `Double` and `Int` at least? Again, having `Evidence` sealed makes it very unpractical to work with these. And I think interpreting the primitive values as `js.Any` is pretty common (or is that just our code?).